app/services/punches_service.rb

```diff
@@ -0,0 +1,20 @@
+class PunchesService
```

Comment:
What do you think about naming this service `CreatePunchService`? It sounds more like a service object.

Reply:
I didnt know in the ruby if a service do one thing or several related with the model

Reply:
Ideally your service is going to perform a single action, that's the reason in general you'll see the service name using the verb as a prefix, `Create<something>`, `Update<something>`

Reply:
I get it, the problem was in this repo don't have much services with this action prefix